pkg/cli/term/setup_windows.go

```diff
@@ -34,6 +43,22 @@ func setup(in, out *os.File) (func() error, error) {
 	errSetOut := windows.SetConsoleMode(hOut, wantedOutMode)
 	errVT := setupVT(out)
 
+	layout, _, err := syscall.Syscall(procGetKeyboardLayout.Addr(), 1, 0, 0, 0)
+	if err == windows.ERROR_SUCCESS {
+
+		// Shamelessly stolen from
+		// https://stackoverflow.com/questions/54588823/detect-if-the-keyboard-layout-has-altgr-on-it-under-windows
+		for char := 0x20; char <= 0xff; char += 1 {
+			scancode, _, err := syscall.Syscall(procVkKeyScanExA.Addr(), 2, uintptr(char), layout, 0)
+
+			if err == windows.ERROR_SUCCESS && scancode&0x0600 == 0x0600 {
+				// At least one ASCII char requires CTRL and ALT to be pressed
+				currentKeyboardLayoutHasAltGr = true
+				break
+			}
+		}
+	}
+
```

Comment:
I don't think Elvish should assume that the user never changes the keyboard layout during a shell session. I believe the check here is unnecessary too; see my other comment.

Reply:
Handling AltGr correctly does not require detecting the layout. I won't go into too much detail here, but this should be quite clear once you understand how reader_windows.go parses the event.